### ISSUE TITLE
Don't overload show(io, ::Type)

### ISSUE THREAD

**Issue by ararslan** [invenia/Intervals.jl#31]
This is type treason, as Jeff puts it, and makes Bad Things Happen on 0.7. See https://github.com/JuliaLang/julia/issues/24195. If we simply do away with those methods, the printing is a little less concise, but still reasonable (IMO):
```
julia> HourBeginning
AnchoredInterval{1 hour,T} where T<:Dates.TimeType
```
This is in line with how Base prints aliases, e.g.
```
julia> AbstractVecOrMat
Union{AbstractArray{T,1}, AbstractArray{T,2}} where T
```
Thoughts?

**Comment by rofinn** [comment]
I'd be fine with that change. This should also be a little simpler once we merge #12.

**Comment by iamed2** [comment]
Is it possible to override only how they're shown in the REPL? Could we override just 
`show(::IO, ::MIME"text/plain", ::Type{T}) where T <: ...`

**Comment by iamed2** [comment]
Or `display`?

**Comment by ararslan** [comment]
Either way we're still committing type piracy though.

**Comment by iamed2** [comment]
It sounds like we have to do this, so let's add `showarg` methods at the same time to mitigate the downside.

**Comment by ararslan** [comment]
Closed by #30.
